src/browser/modules/Sidebar/ProjectsFilesScripts.tsx

```diff
@@ -0,0 +1,213 @@
+  GET_PROJECT_FILES,
+  DELETE_PROJECT_FILE,
+  REMOVE_PROJECT_FILE
+} from './project-files.constants'
+import Render from 'browser-components/Render'
+import { Bus } from 'suber'
+
+interface IProjectFilesError {
+  apolloErrors: (ApolloError | undefined)[]
+  errors?: string[]
+}
+
+export const ProjectFilesError = ({
+  apolloErrors,
+  errors
+}: IProjectFilesError): JSX.Element => {
+  const definedApolloErrors = filter(
+    apolloErrors,
+    apolloError => apolloError !== undefined
+  )
+  const definedErrors = filter(errors, error => error !== '')
+  const ErrorsList = () => {
+    if (size(definedErrors)) {
+      return definedErrors.map((definedError, i) => (
+        <span key={`${definedError}-${i}`}>{definedError}</span>
+      ))
+    }
+    if (size(definedApolloErrors)) {
+      return definedApolloErrors.map((definedApolloError, j) => {
+        if (definedApolloError?.graphQLErrors.length) {
+          return definedApolloError.graphQLErrors.map(({ message }, k) => (
+            <span key={`${message}-${j}-${k}`}>{message}</span>
+          ))
+        }
+        if (definedApolloError?.networkError) {
+          return <span key={j}>A network error has occurred.</span>
+        }
+        return null
+      })
+    }
+    return null
+  }
+
+  return (
+    <Render if={size(definedApolloErrors) || size(definedErrors)}>
+      {ErrorsList()}
+    </Render>
+  )
+}
+
+interface IProjectFilesScripts {
+  bus: Bus
+  onSelectScript: (favorite: IFavorite) => void
+  onExecScript: (favorite: IFavorite) => void
+  onExportScripts: () => void
+  onUpdateFolder: () => void
+  onRemoveFolder: () => void
+  scriptsNamespace: string
+  title: string
+}
+
+function ProjectFilesScripts(props: IProjectFilesScripts): JSX.Element {
+  const { data, error: getProjectFilesError, refetch } = useQuery<
+    IProjectFilesResult,
+    IProjectFilesVariables
+  >(GET_PROJECT_FILES, {
+    variables: ProjectFilesQueryVars
+  })
+  const [removeFile, { error: removeProjectFileError }] = useMutation(
+    DELETE_PROJECT_FILE
+  )
+  const [projectFiles, setProjectFiles] = useState<IFavorite[]>([])
+
+  useEffect(() => {
+    let isStillMounted = true
+    if (data) {
+      const getProjectFiles = async () => {
+        const getProjectFilePromises = data.getProject.files.map(
+          mapProjectFileToFavorites
+        )
+        const projectFiles = await Promise.all(getProjectFilePromises)
+        if (isStillMounted) {
+          setProjectFiles(projectFiles)
+        }
+      }
+      getProjectFiles()
+    }
+    return () => {
+      isStillMounted = false
+    }
+  }, [data])
+
+  useEffect(() => {
+    if (data && refetch) {
+      refetch()
+    }
+  }, [data, refetch])
+
+  const myScriptsProps = {
+    ...omit(props, 'bus'),
+    scripts: projectFiles,
+    isProjectFiles: true,
+    onRemoveScript: async (favorite: IFavorite) => {
+      const directory = favorite.path.substring(1) // @todo: adding in SLASH to path
```

Comment:
for next pr?

Reply:
I might just change it to a comment and leave it. Its not a blocker

Reply:
Will change in my next PR, want to double check it all